react/features/filmstrip/functions.any.js

```diff
@@ -50,13 +64,32 @@ export function updateRemoteParticipants(store: Object, participantId: ?number)
         remoteParticipants.delete(speaker);
     }
 
-    // Always update the order of the thumnails.
-    reorderedParticipants = [
-        ...Array.from(screenShares.keys()),
-        ...sharedVideos,
-        ...Array.from(speakers.keys()),
-        ...Array.from(remoteParticipants.keys())
-    ];
+    if (getSourceNameSignalingFeatureFlag(state)) {
```

Comment:
The participants with screenshare needs to be pushed to the top of the remote participants list which doesn't seem to be happening here.

Reply:
Also, if there is a local fakeScreenShareParticipant, it needs to be pushed to the top so that it appears as the second tile after the local participant's video preview tile (in both tile view and stage view)